clients/privacy-center/cypress/e2e/consent-banner-tcf.cy.ts

```diff
@@ -722,7 +724,83 @@ describe("Fides-js TCF", () => {
           ).to.eql(true);
         });
       });
-
+      it("skips saving preferences to API when custom API fn is provided in Fides.init", () => {
+        cy.fixture("consent/experience_tcf.json").then((privacyExperience) => {
+          stubConfig({
+            options: {
+              isOverlayEnabled: true,
+              tcfEnabled: true,
+              api: {
+                /* eslint-disable @typescript-eslint/no-unused-vars */
+                savePreferencesFn: (
+                  fides: FidesConfig,
+                  consent: CookieKeyConsent,
+                  fides_string: string | undefined,
+                  experience: PrivacyExperience
+                ) => {},
+                /* eslint-enable @typescript-eslint/no-unused-vars */
+              },
+            },
+            experience: privacyExperience.items[0],
+          });
+        });
+        cy.waitUntilFidesInitialized().then(() => {
+          cy.get("#fides-modal-link").click();
+          cy.getByTestId("consent-modal").within(() => {
+            cy.get("button").contains("Opt out of all").click();
+            // timeout means API call not made, which is expected
+            cy.on("fail", (error) => {
+              if (error.message.indexOf("Timed out retrying") !== 0) {
+                throw error;
+              }
+            });
```

Comment:
This is a pattern we use for other tests that require asserting a certain cypress mock was not called. Basically, we set a timeout on the call we `wait` for. That call never happens, as expected, so Cypress triggers a `fail`. 

This function intercepts the Cypress `fail`, and if we don't manually `throw error`, then Cypress won't actually fail the test. So in this case, we still want Cypress to fail the test for any other reason, _except_ when we have a timeout.

 The downfall of this is that we could have a false positive for tests that timeout for other reasons than we intended, so that's why we keep this intercept scoped to just 1 specific assertion.

Reply:
OK, gotcha- I'm remembering this now... thanks